### PR TITLE
Bw/learning mfe updates

### PR DESCRIPTION
**Pull request by muselesscreator**
A couple of changes to support test in the learning MFE:

update getEffect to getEffects to catch multiple effects w/ same prereqs.
Pass in reactModule to getEffects to enable cross-repo mocking
add mockVals and resetVals in mockUseKeyedState to enable mocking multiple values for a test.

### CODE REVIEW

File: src/testUtils/shallow.jsx [hunk 1]
```diff
@@ -28,9 +28,11 @@ class ReactShallowRenderer {
     try {
       out = new ReactShallowRenderer(Component);
     } catch (error) {
-      console.log({ error, Component });
+      if (!error.message.includes('ReactShallowRenderer')) {
```
Comment (leangseu-edx) [inline src/testUtils/shallow.jsx:31]: What is the reason for this?

Comment (muselesscreator) [reply src/testUtils/shallow.jsx:31]: So, Shallow Rendering doesn't work on basic jsx (`<div><h1>Hi</h1></div>`), and when we try and it fails for that reason, we provide a flat render.  *BUT* we don't want to swallow other errors.  This just swallows the "shallow rendering doesn't work on jsx" error pretty much, while allowing errors in the code/test to raise errors appropriately.

File: src/testUtils/getEffects.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * getEffect(prereqs, reactModule)
+ * Returns the callbacks passed to useEffect mock call with the given prerequisites key,
+ * or null if none match.
+ *
+ * @param {any[]} prereqs - unique prerequisite list identifying the target useEffect call
+ * @param {object} reactModule - react module to be mocked.
+ * @return {function} - callback method passed for the effect hook.
+ */
+const getEffects = (prereqs, reactModule) => {
```
Comment (leangseu-edx) [inline src/testUtils/getEffects.js:11]: why passing react module as arguments? Can we just create a method that `setupTest.js` can invoke to mock React?

Comment (muselesscreator) [reply src/testUtils/getEffects.js:11]: Alas, this repo will not run setupTest against the parent, and attempting to mock from the local version of React did not override for the tests.

Comment (leangseu-edx) [reply src/testUtils/getEffects.js:11]: I mean we create a method like `setupReactUtilTest()` that can get import on `setupTest.js`.

Comment (muselesscreator) [reply src/testUtils/getEffects.js:11]: but that requires it hit the whole repo at once

**Review by leangseu-edx — APPROVED** [review]

**Pull request merged**